File: Bitmask/1052번 - 물병/1052.cpp
#include <iostream>
#include <bitset>
#include <algorithm>
#include <vector>

using namespace std;

int n, k;
vector<int> v;

void input() {
    cin >> n >> k;
}

void solve() {
    int ans = 0;
    
    while(1) {
        // ans는 추가 물병 개수
        int tmp = n + ans;
        int cnt = 0;
        
        // tmp의 값을 2진수로 1의 개수를 파악
        while(tmp) {
            if(tmp & 1) cnt++;
            tmp >>= 1;
        }
        
        if(cnt <= k) break;
        ans++;
    }
    
    cout << ans << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Bitmask/1052번 - 물병/1052.java
import java.io.*;
import java.util.*;

public class Main {
    static int n;
    static int k;

    public static void main(String[] args) throws IOException {
        input();
        solve();
    }

    static void input() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        k = Integer.parseInt(st.nextToken());
    }

    static void solve() {
        int ans = 0;
        while (countBits(n + ans) > k) {
            ans++;
        }
        System.out.println(ans);
    }

    static int countBits(int x) {
        int cnt = 0;
        while (x != 0) {
            if ((x & 1) == 1) cnt++;
            x >>= 1;
        }
        return cnt;
    }
}